### PR TITLE
feat: accordion menu mobile only

### PR DESCRIPTION
**Pull request by boyarskiy**
https://autoricardo.atlassian.net/browse/VSST-421

## Motivation and context

Show accordion only on mobile

## Before

No accordion only on mobile

## After

md
Screenshot 2022-11-02 at 07 43 53

sm
Screenshot 2022-11-02 at 07 44 16

## How to test

https://boyarskiy-accordion-menu.components-pkg.branch.autoscout24.ch/?path=/story/patterns-navigation-accordion-accordionmenu--accordion-menu-mobile-only

### CODE REVIEW

File: src/components/index.ts [hunk 1]
```diff
@@ -33,6 +33,8 @@ export { default as Tooltip } from './tooltip';
 export { default as Accordion } from './accordion';
 export { default as AccordionButton } from './accordion/AccordionButton';
 export { default as AccordionPanel } from './accordion/AccordionPanel';
+export { default as AccordionMenu } from './accordion/AccordionMenu';
```
Comment (talamcol) [inline src/components/index.ts:36]: shouldn't this component only be used inside components-pkg for the header/footer? I wouldn't export it if that is the case

File: src/components/accordion/AccordionMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { FC, PropsWithChildren } from 'react';
+
+import { AccordionProps } from '@chakra-ui/react';
```
Comment (talamcol) [inline src/components/accordion/AccordionMenu.tsx:3]: is it needed for this component to have full flexibility or can we restrict the props to some subset?

File: src/components/accordion/AccordionMenuSection.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+const AccordionMenuSection: FC<PropsWithChildren<AccordionMenuItemProps>> = (
+  props
+) => {
+  const { children, title, ...itemProps } = props;
+
+  return (
+    <>
+      <Show below="sm">
```
Comment (talamcol) [inline src/components/accordion/AccordionMenuSection.tsx:25]: I find this a bit weird to be in a component that is exported. The developer that uses this component doesn't know it's mobile-only without checking the implementation detail. I would expect this logic to be part of the footer component and not an exported one. (we can still have it in a separate file, but I wouldn't export it)

File: src/components/accordion/AccordionMenuSection.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        <AccordionItem {...itemProps}>
+          <AccordionButton>{title}</AccordionButton>
+          <AccordionPanel>{children}</AccordionPanel>
+        </AccordionItem>
+      </Show>
+      <Hide below="sm">
+        <Box paddingY="md" textStyle="heading4">
```
Comment (talamcol) [inline src/components/accordion/AccordionMenuSection.tsx:32]: can we use the H4 component?

File: src/components/accordion/accordionMenu.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+          content: (
+            <List>
+              <ListItem>Item 1</ListItem>
+              <ListItem>Item 2</ListItem>
+              <ListItem>Item 3</ListItem>
+            </List>
+          ),
+        },
+      ],
+    }}
```
Comment (talamcol) [inline src/components/accordion/accordionMenu.stories.mdx:57]: would it be possible to remove the chakra props from the argstable in storybook?

**Review by lkappeler — COMMENTED** [review]
What I had in mind is to create a custom variant https://chakra-ui.com/docs/components/button/theming#adding-a-custom-variant within the select component and not a new component so we can consume it with `Accordion variant="mobileOnly"` since the interface does not change compared to default variation

@talamcol What's your take does that make it more clear in terms of usage?

**Comment by boyarskiy** [comment]
> What I had in mind is to create a custom variant https://chakra-ui.com/docs/components/button/theming#adding-a-custom-variant within the select component and not a new component so we can consume it with `Accordion variant="mobileOnly"` since the interface does not change compared to default variation
> 
> @talamcol What's your take does that make it more clear in terms of usage?

@lkappeler I'm ok to rebuild it with `Accordion variant="mobileOnly"` I agree it indeed might be more clear to use
@talamcol what do you think?

**Comment by talamcol** [comment]
@lkappeler @boyarskiy as discussed yesterday in Slack. I find that a bit misleading because in desktop state it's no longer an accordion. But as you said in Slack, it's only used for the footer which will be in the components-pkg as well. If it's only an internal variant and not exposed in the library, it would be fine I guess. But then we should remove the export

**Comment by boyarskiy** [comment]
Based on discussed with designer we will use this kind of component only in header/footer, so no need to export it.

**Review by talamcol — APPROVED** [review]
thanks for all the extra rounds! Looks good 👍

**Review by lkappeler — APPROVED** [review]
The implementation is looking fine.
I would name the component something like `MobileOnlyAccordion` to describe what it does rather than where we will use it. (Also we have a Menu component which behaves differently that we will use in the header and footer)

**Comment by boyarskiy** [comment]
> The implementation is looking fine. I would name the component something like `MobileOnlyAccordion` to describe what it does rather than where we will use it. (Also we have a Menu component which behaves differently that we will use in the header and footer)

This name is fine, renamed one more time :grinning:

**Pull request merged**